pkg/deploy/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -125,6 +148,15 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 				if err != nil {
 					return err
 				}
+				// We need to make sure we own that RC or adopt it if possible
+				isOurs, err := cm.ClaimReplicationController(rc)
```

Comment:
@kargakis the one we get with the `Get` could be a different one. What if it no longer matches the selector and was/should be released or it could have been changed, deleted/+created and we don't own it